azurelinuxagent/ga/exthandlers.py

```diff
@@ -278,15 +278,29 @@ def get_goal_state_debug_metadata(self):
         :return: Tuple (activity_id, correlation_id, gs_created_timestamp) or "NA" for any property that's not available
         """
 
-        def check_empty(value): return value if value not in (None, "") else "NA"
+        def parse_value(parse_fn, value):
+
+            try:
+                if value not in (None, ""):
+                    return parse_fn(value)
+            except Exception as e:
+                # A failure here isn't a fatal error, because the info we're
+                # trying to retrieve is debug only on linux.
+                error_msg = u"Couldn't parse debug metadata value: {0}".format(e)
+                logger.verbose(error_msg)
+            
+            return "NA"
```

Comment:
Rather than returning for None, empty string or unparsable time we should just leave the value unchanged so that we know what we are getting from CRP. Displaying it as a date/time was just meant to make reading the log easier. If we get "0", we should just log "0".